platforms/chibios/boards/GENERIC_STM32_G431XB/configs/hal_lld.h

```diff
@@ -0,0 +1,8 @@
+// Copyright 2024 QMK
+// SPDX-License-Identifier: GPL-2.0-or-later
+#pragma 
+
+// TODO: Remove when upstream no longer requires patching
+#include "stm32_registry.h"
```

Comment:
```suggestion
#include <stm32_registry.h>
```
Using double quotes here breaks the workaround (you may notice that `handwired/onekey/nucleo_g431rb` does not build).  The reason is that using double quotes effectively prepends the directory of the currently compiled file to the include path, therefore `#include_next` in the overridden `stm32_registry.h` skips just that entry in the modified include path, and finds itself the second time instead of finding the intended copy in ChibiOS.

(An ancient discussion of that behavior is https://gcc.gnu.org/legacy-ml/gcc-patches/2005-03/msg01723.html — the patch changing that behavior was rejected, because the real problem is using `#include "foo.h"` where `<foo.h>` was intended; in fact, we are working around exactly that problem in ChibiOS itself by intercepting an earlier include that does not get resolved to a file in the same directory as the including one).